tests/end-to-end/director/runtime_auth_test.go

```diff
@@ -0,0 +1,149 @@
+				{
+					Name:      "without-default-auth",
+					TargetURL: "http://mywordpress.com/comments",
+				},
+			},
+		},
+		{
+			Name: "API with default auth",
+			Apis: []*graphql.APIDefinitionInput{
+				{
+					Name:        "with-default-auth",
+					TargetURL:   "http://mywordpress.com/comments",
+					DefaultAuth: defaultAuth,
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			appInput := graphql.ApplicationInput{
+				Name: "test-app",
+				Apis: testCase.Apis,
+				Labels: &graphql.Labels{
+					"scenarios": []interface{}{"DEFAULT"},
+				},
+			}
+
+			app := createApplicationFromInputWithinTenant(t, ctx, appInput, defaultTenant)
+			defer deleteApplication(t, app.ID)
+
+			var rtmIDs []string
+			for i := 0; i < 10; i++ {
+				rtm := createRuntime(t, ctx, fmt.Sprintf("test-rtm-%d", i))
+				rtmIDs = append(rtmIDs, rtm.ID)
+			}
+			defer func() {
+				for _, id := range rtmIDs {
+					deleteRuntime(t, id)
+				}
+			}()
+			require.True(t, len(rtmIDs) > 1)
+
+			rtmIDWithUnsetRtmAuth := rtmIDs[0]
+			rtmIDWithSetRtmAuth := rtmIDs[1]
+
+			setAPIAuth(t, ctx, app.Apis.Data[0].ID, rtmIDWithSetRtmAuth, customAuth)
+			defer deleteAPIAuth(t, ctx, app.Apis.Data[0].ID, rtmIDWithSetRtmAuth)
+
+			innerTestCases := []struct {
+				Name         string
+				QueriedRtmID string
+			}{
+				{
+					Name:         "Query set Runtime Auth",
+					QueriedRtmID: rtmIDWithSetRtmAuth,
+				},
+				{
+					Name:         "Query unset Runtime Auth",
+					QueriedRtmID: rtmIDWithUnsetRtmAuth,
+				},
+			}
+
+			for _, innerTestCase := range innerTestCases {
+				t.Run(innerTestCase.Name, func(t *testing.T) {
+					result := ApplicationExt{}
+
+					request := gcli.NewRequest(
+						fmt.Sprintf(`query {
+							result: application(id: "%s") {
+								%s
+							}
+						}`, app.ID, tc.gqlFieldsProvider.ForApplication(fieldCtx{
+							"APIDefinition.auth": fmt.Sprintf(`auth(runtimeID: "%s") {%s}`, innerTestCase.QueriedRtmID, tc.gqlFieldsProvider.ForRuntimeAuth()),
+						})))
+
+					// WHEN
+					err := tc.RunQuery(ctx, request, &result)
+
+					// THEN
+					require.NoError(t, err)
+					require.NotEmpty(t, result.ID)
+					assertApplication(t, appInput, result)
+					require.Equal(t, 1, len(result.Apis.Data))
+
+					assert.Equal(t, len(rtmIDs), len(result.Apis.Data[0].Auths))
+					assert.Equal(t, innerTestCase.QueriedRtmID, result.Apis.Data[0].Auth.RuntimeID)
+					if innerTestCase.QueriedRtmID == rtmIDWithSetRtmAuth {
+						assertAuth(t, &customAuth, result.Apis.Data[0].Auth.Auth)
+					} else {
+						assert.Equal(t, result.Apis.Data[0].DefaultAuth, result.Apis.Data[0].Auth.Auth)
+					}
+
+					for _, auth := range result.Apis.Data[0].Auths {
+						if auth.RuntimeID == rtmIDWithSetRtmAuth {
+							assertAuth(t, &customAuth, auth.Auth)
+						} else {
+							assert.Equal(t, result.Apis.Data[0].DefaultAuth, auth.Auth)
+						}
+					}
+
+					if !exampleSaved {
```

Comment:
Use `innerTestCase.QueriedRtmID == rtmIDWithSetRtmAuth` instead of this.

Reply:
`QueriedRtmID` will have that value two times, I wanted to save the example only once.